livesync_citadel/indico_livesync_citadel/schemas.py

```diff
@@ -0,0 +1,263 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2020 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import itertools
+from marshmallow import post_dump
+from tika import parser
+from webargs import fields
+
+from indico.core.db.sqlalchemy.links import LinkType
+from indico.core.db.sqlalchemy.principals import PrincipalType
+from indico.core.db.sqlalchemy.protection import ProtectionMode
+from indico.core.marshmallow import mm
+from indico.modules.attachments.models.attachments import Attachment, AttachmentType
+from indico.modules.events import Event
+from indico.modules.events.contributions.models.contributions import Contribution
+from indico.modules.events.contributions.models.subcontributions import SubContribution
+from indico.modules.events.notes.models.notes import EventNote
+from indico.web.flask.util import url_for
+
+
+def _get_identifiers(principal):
+    if (principal.principal_type == PrincipalType.user
+            or principal.principal_type == PrincipalType.event_role
+            or principal.principal_type == PrincipalType.category_role):
+        yield principal.identifier
+        yield principal.email
+    elif principal.is_group:
+        yield principal.identifier
+
+
+class ACLSchema(object):
+    _access = fields.Method('_get_object_acl')
+
+    def _get_event_acl(self, event):
+        if event.effective_protection_mode == ProtectionMode.public:
+            return []
+        return set(itertools.chain.from_iterable(_get_identifiers(x.principal) for x in event.acl_entries))
```

Comment:
@ThiefMaster should this also return the category managers that access to the event?

Reply:
Yes, you can use the `get_access_list()` method for this. This is what we also do when building the legacy MARCXML data for the old search.